src/test/java/com/google/sps/UserTripServletTest.java

```diff
@@ -0,0 +1,166 @@
+
+package com.google.sps;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.Config;
+import com.google.sps.data.User;
+import com.google.sps.servlets.AuthServlet;
+import com.google.sps.servlets.UserTripServlet;
+import com.google.sps.servlets.TripServlet;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(UserServiceFactory.class)
+public final class UserTripServletTest {
+
+  // Add constants necessary to testing retrieval of the current User.
+  public static final String EMAIL = "testemail@gmail.com";
+  public static final String AUTH_DOMAIN = "gmail.com";
+  public static final String LOGOUT_URL = "/_ah/logout?continue=%2F";
+  public static final String LOGIN_URL = "/_ah/login?continue=%2F";
+
+  // Create UserTripServlet and TripServlet objects.
+  UserTripServlet userTripServlet;
+  TripServlet tripServlet;
+
+  // Add helper to allow datastore testing in local JUnit tests.
+  // See https://cloud.google.com/appengine/docs/standard/java/tools/localunittesting.
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void initServlets() {
+    userTripServlet = new UserTripServlet();
+    tripServlet = new TripServlet();
+  }
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testDoGetResponseLoggedIn() throws Exception {
+    // Mock request and response.  
+    HttpServletRequest requestMock = mock(HttpServletRequest.class);    
+    HttpServletResponse responseMock = mock(HttpServletResponse.class);
+    
+    // Create Trip Entity properties.
+    final String tripName = "Family Vacation";
+    final String destinationName = "Island of Hawai'i";
+    final String tripDayOfTravel = "2020-07-17";
+    final String photoSrc = "../images/placeholder_image.png";
+
+    // Mock UserService methods as logged-in user.
+    UserService userServiceMock = mock(UserService.class);
+    when(userServiceMock.isUserLoggedIn()).thenReturn(true);
+    // This is the User object from Google Appengine (full path given to avoid
```

Comment:
Jut to be consistent with the rest of your code add a line between lines 91 and 92

Reply:
Done!